src/nvim/statusline.c

```diff
@@ -73,9 +73,17 @@ void win_redr_status(win_T *wp)
   // invokes ":redrawstatus".  Simply ignore the call then.
   if (busy
       // Also ignore if wildmenu is showing.
-      || (wild_menu_showing != 0 && !ui_has(kUIWildmenu))) {
+      || (wild_menu_showing != 0 && !ui_has(kUIWildmenu))
+      // Floating windows don't show statusline with global statusline
+      || (wp->w_floating && is_stl_global)) {
     return;
   }
+  wp->w_redr_status = false;
+
+  if (wp->w_floating && is_stl_global) {
+    return;
```

Comment:
?

If `wp->w_redr_status = false;` is needed, then line 78 should be dropped. 
If `wp->w_redr_status = false;` is NOT needed, then line 83 should be dropped.

Right?